M/MongoC/build_tarballs.jl

```diff
@@ -0,0 +1,48 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "MongoC"
+version = v"1.16.2"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://github.com/mongodb/mongo-c-driver/releases/download/1.16.2/mongo-c-driver-1.16.2.tar.gz", "0a722180e5b5c86c415b9256d753b2d5552901dc5d95c9f022072c3cd336887e")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd mongo-c-driver-1.16.2/
+sed -i "s/Windows.h/windows.h/" src/libmongoc/src/mongoc/mongoc-client.c
+sed -i "s/WinDNS.h/windns.h/" src/libmongoc/src/mongoc/mongoc-client.c
+sed -i "s/Mstcpip.h/mstcpip.h/" src/libmongoc/src/mongoc/mongoc-client.c
+sed -i "s/Mstcpip.h/mstcpip.h/" src/libmongoc/src/mongoc/mongoc-socket.c
+sed -i "s/Dnsapi/dnsapi/" build/cmake/FindResSearch.cmake
+mkdir cmake-build
+cd cmake-build/
+cmake  -DENABLE_AUTOMATIC_INIT_AND_CLEANUP=OFF -DCMAKE_BUILD_TYPE=Release -DENABLE_SSL=OPENSSL -DENABLE_SASL=OFF -DENABLE_EXAMPLES=OFF -DENABLE_TESTS=OFF -DENABLE-UNINSTALL=OFF -DENABLE_STATIC=OFF -DENABLE_MONGOC=ON -DENABLE_BSON=ON -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} ..
+make -j${nproc}
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libbson-1.0", :libbson),
+    LibraryProduct("libmongoc-1.0", :libmongoc),
```

Comment:
You can use `Pkg.BinaryPlatforms.parse_dl_name_version` to find what's the name expected:
```julia
julia> using Pkg.BinaryPlatforms

julia> parse_dl_name_version("bin/libbson-1.0.dll", Windows(:x86_64))
("libbson", v"1.0.0")
```

For macOS the situation is quite fascinating: the library is created with this ridiculous file name: `lib/libbson-1.0.0.0.0.dylib`.  If you try to use `parse_dl_name_version` you get this:
```julia
julia> parse_dl_name_version("lib/libbson-1.0.0.0.0.dylib", MacOS(:x86_64))
ERROR: ArgumentError: invalid version string: 0.0.0.0
Stacktrace:
 [1] VersionNumber(::SubString{String}) at ./version.jl:111
 [2] parse_dl_name_version(::String, ::MacOS) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/Pkg/src/BinaryPlatforms.jl:653
 [3] top-level scope at REPL[9]:1
 [4] eval(::Module, ::Any) at ./boot.jl:331
 [5] eval_user_input(::Any, ::REPL.REPLBackend) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/REPL/src/REPL.jl:86
 [6] run_backend(::REPL.REPLBackend) at /home/mose/.julia/packages/Revise/kqqw8/src/Revise.jl:1163
 [7] top-level scope at none:0
```

It's probably better to rename the file to something more sensible like `libbson.1.0.dylib`:
```julia
julia> parse_dl_name_version("lib/libbson.1.0.dylib", MacOS(:x86_64))
("libbson", v"1.0.0")
```

Reply:
Sorry, I don't understand where this `parse_dl...` check should go.

Reply:
You don't need to call `parse_dl_name_version` directly in the script, but its output tells you how the auditor is parsing the names of the files on disk and why the name you used in the script is not being accepted.  At the beginning there was `libbson`, but the the library on Linux was built with name `lib/libbson-1.0.so`, that's why I suggested to change it here to `libbson-1.0`:
```
julia> parse_dl_name_version("lib/libbson-1.0.so", Linux(:x86_64))
("libbson-1.0", nothing)
```

Now we have to find the names that will match what the auditor will parse.  For Windows we need
```suggestion
    LibraryProduct(["libbson-1.0", "libbson"], :libbson),
    LibraryProduct(["libmongoc-1.0", "libmongoc"], :libmongoc),
```
for macOS I suggest to rename the libraries to `lib....1.0.dylib` at the end of the script, so that they'll be parsed as
```julia
julia> parse_dl_name_version("lib/libbson.1.0.dylib", MacOS(:x86_64))
("libbson", v"1.0.0")
```